neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/data/im/InMemoryBackendFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2013-2017 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.data.im;
+
+import fr.inria.atlanmod.commons.annotation.Static;
+import fr.inria.atlanmod.neoemf.config.ImmutableConfig;
+import fr.inria.atlanmod.neoemf.data.AbstractBackendFactory;
+import fr.inria.atlanmod.neoemf.data.Backend;
+import fr.inria.atlanmod.neoemf.data.BackendFactory;
+
+import org.eclipse.emf.common.util.URI;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * A factory that create {@link DefaultInMemoryBackend} instances.
+ */
+@ParametersAreNonnullByDefault
+public class InMemoryBackendFactory extends AbstractBackendFactory {
+
+    /**
+     * The literal description of the factory.
+     */
+    public static final String NAME = "im"; // In-Memory
+
+    /**
+     * Constructs a new {@code InMemoryBackendFactory}.
+     */
+    protected InMemoryBackendFactory() {
+        super();
+    }
+
+    /**
+     * Returns the instance of this class.
+     *
+     * @return the instance of this class
+     */
+    @Nonnull
+    public static BackendFactory getInstance() {
+        return Holder.INSTANCE;
+    }
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public boolean supportsTransient() {
+        return true;
+    }
+
+    @Nonnull
+    @Override
+    public Backend createBackend(URI uri, ImmutableConfig baseConfig) {
+        return new DefaultInMemoryBackend();
+    }
+
+    /**
+     * The initialization-on-demand holder of the singleton of this class.
+     */
+    @Static
+    private static final class Holder {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)